ncm-spma/src/main/perl/spma/ips.pm

```diff
@@ -451,131 +498,189 @@ sub update_ips
         return 0;
     }
 
+    my $exact_install;
+    unless ($allow_user_pkgs or scalar(%$uninst)) {
+        #
+        # Determine whether this version of Solaris is new enough to support
+        # the pkg exact-install command, which speeds this process up
+        #
+        $exact_install = $self->pkg_has_exact_install();
+    } else {
+        #
+        # If we are allowing user packages, we will not use pkg exact-install
```

Comment:
maybe better to handle each case in it own if/elif/else block, and report this comment with `verbose`.